File: MoviesAPIClient/Features/DisplayServer/DisplayServer.swift
//
//  DisplayServer.swift
//  MoviesAPIClient
//
//  Created by Aryan Rogye on 9/15/26.
//

import Foundation

enum DisplayServerError: LocalizedError {
    case cantConstructURL

    var errorDescription: String? {
        switch self {
        case .cantConstructURL:
            return "Can't construct URL"
        }
    }
}

enum DisplayServer: String, CaseIterable {
    case moviesAPI = "MovesAPI"
    case vidFast = "VidFast"

    var baseURL: URL? {
        switch self {
        case .moviesAPI:
            URL(string: "https://moviesapi.to")
        case .vidFast:
            URL(string: "https://vidfast.vc")
        }
    }

    public func loadMovie(
        movieId: Int
    ) throws -> URL {

        guard let baseURL else {
            throw DisplayServerError.cantConstructURL
        }

        let url = baseURL
            .appendingPathComponent("movie")
            .appendingPathComponent(String(movieId))

        return url
    }

    public func loadTvShow(
        showId: Int,
        season: Int,
        episode: Int
    ) throws -> URL {
        guard let baseURL else {
            throw DisplayServerError.cantConstructURL
        }

        let url = baseURL
            .appendingPathComponent("tv")
            .appendingPathComponent(String(showId))
            .appendingPathComponent(String(season))
            .appendingPathComponent(String(episode))

        return url
    }
}

File: MoviesShared/sharedLogic/src/commonMain/kotlin/com/aryanrogye/movies_shared/network/DisplayServer.kt
package com.aryanrogye.movies_shared.network

enum class KTDisplayServer(val rawValue: String) {
    MOVIES_API("MoviesAPI"),
    VID_FAST("VidFast"),

    VID_SPARK("VidSpark"),

    VID_LINK("VidLink");

    private val baseURL: String
        get() {
            return when (this) {
                MOVIES_API -> "https://moviesapi.to"
                VID_FAST -> "https://vidfast.vc"
                VID_SPARK -> "https://vidspark.to"
                VID_LINK -> "https://vidlink.pro"
            }
        }

    fun loadMovie(movieId: Int): String {
        return "$baseURL/movie/$movieId"
    }

    fun loadTvShow(showId: Int, season: Int, episode: Int): String {
        return "$baseURL/tv/$showId/$season/$episode"
    }
}